lib/css_injector.js

```diff
@@ -14,13 +14,48 @@ CSSInjector.prototype = {
   write: function(outputPath) {
     let inputHTML = fs.readFileSync(path.join(this.rootPath, 'index.html'), 'utf-8');
 
-    this.filePathsToInject.forEach((p) => {
-      if (inputHTML.indexOf(p) === -1) { return; }
+    let generatedCss = [];
+    this.filePathsToInject.forEach((filePath) => {
+      let pathIndex = filePath.lastIndexOf('/');
+      let assetPath = filePath.substr(0, pathIndex);
+      let fileName = filePath.substr(pathIndex + 1) || '';
+      fileName = fileName.replace('.css', '');
+      let rootPath = `${this.rootPath}${assetPath}`;
 
-      let regex = new RegExp(`<link[^>]* href="[^"]*${p}"[^>]*>`)
+      /* RegExp to match both fingerprinted and non fingerprinted file names */
+      let nameRegex = new RegExp(`(${fileName}.css|${fileName}-\\b[0-9a-f]{5,40}\\b.css)`);
+      let matchedFile = '';
 
+      /* Generates a array css files from the output directory.
+       * Directory operation is done only once.
+       * Later this `generatedCss` array is iterated to match the fileName.
+       */
+      if (generatedCss.length === 0) {
```

Comment:
@mixonic This was added to limit the directory operation to be done only once. But this can be wrong at times.

Eg: Say we get custom filter options.

```
'ember-inline-css': {
   filter: ['/assets/vendor.css', '/assets/styles/app.css']
}
```

In the above example `/assets/vendor.css` is by mistake specified as a wrong path. We will now look for the generated css under `/assets/` directory (wrong directory) and generate it if there are any files available there. We will not replace `vendor.css` as the wrong path is specified, that is perfectly fine. But we will also not replace the `app.css` even though the specified path is valid for it. We will not assert or throw any error here, just the files wouldn't have been replaced.

What should be the take here, can we go with this as a known behaviour? Or should we do directory operation every time to fetch the correct output file?